docs/src/user/gotchas.md

```diff
@@ -1,13 +1,69 @@
 # Gotchas
 
-Here are some potential "gotchas" for interpreter authors:
+Yk has some unusual implications, requirements and limitations that interpreter
+authors should be aware of.
+
+## Trace quality depends upon completeness of interpreter IR.
+
+Yk works by recording control flow paths (traces) through your interpreter
+implementation. Each trace ends up as an ordered list of LLVM IR blocks which
+are "stitched together" and compiled. During the trace collection process, by
```

Comment:
I'd replace the last sentence with something like "Unless functions are marked yk_noinline, they will be inlined into the trace".